### PR TITLE
Add event for response and logging

### LINKED ISSUE #72: Add logging to req/res details for debug mode
To make it easier to debug issues, we should add a log option for req/res details, along with response times.  This should be logged to stdout or if the debug environment variable indicates, to a file.

### PR DESCRIPTION
**Pull request by geek**
Fixes #72

This depends on the fine work @danielb2 did in PR #74

### CODE REVIEW

**Comment by danielb2** [comment]
Very nice approach Wyatt ;)

**Comment by geek** [comment]
@hueniverse thoughts on this approach?  Let the implementer decide on logging, but provide an event for all wreck responses?

File: README.md [hunk 1]
```diff
@@ -195,3 +195,12 @@ For example, the following code demonstrates changing `maxSockets` on the `http`
 
  Wreck.agents.http.maxSockets = 20;
  ```
+
+
+### Events
+
+#### `response`
+
+The response event is always executed for any request that *wreck* makes.  The handler should accept the following
```
Comment (hueniverse) [inline README.md]: s/executed/emitted

File: lib/index.js [hunk 2]
```diff
@@ -17,14 +18,17 @@ var Tap = require('./tap');
 var internals = {};
 
 
-exports.agents = {
+module.exports = new Events.EventEmitter();
```
Comment (hueniverse) [inline lib/index.js]: This is kinda hackish. Better to use normal prototype style and new an object and export it.

**Comment by hueniverse** [comment]
I don't think this is a good idea in its current form. It would be impossible to know what's going on if wreck is shared (because of node_modules) among multiple clients. I am not sure what's the use case here but I would assume if a module uses wreck and wants events, it only wants events triggered by it. So maybe allow using the global instance but also to new another which does not share the same emitter? Or only support events when a new instance is created?

**Comment by geek** [comment]
@hueniverse thanks for the feedback.  I will incorporate your advice.  The major use case is to allow for developers to log wreck requests/responses in one spot.  This is in replace of https://github.com/hapijs/wreck/pull/76

**Comment by geek** [comment]
@hueniverse thoughts on the current PR?  I want to wait and see if anyone has issues with a module wide event emitter before exporting the constructor for Client.

File: lib/index.js [hunk 3]
```diff
@@ -350,49 +363,51 @@ exports.parseCacheControl = function (field) {
...
+internals.Client.prototype.delete = function (uri, options, callback) {
 
-    return internals.shortcutWrap('DELETE', uri, options, callback);
+    return this.shortcutWrap('DELETE', uri, options, callback);
 };
 
 
 // Wrapper so that shortcut can be optimized with required params
-internals.shortcutWrap = function (method, uri /* [options], callback */) {
+internals.Client.prototype.shortcutWrap = function (method, uri /* [options], callback */) {
```
Comment (arb) [inline lib/index.js]: Maybe `_shortcutWarp` since this isn't a documented method?

Comment (danielb2) [inline lib/index.js]: that or make it internals and pass self?

**Pull request merged**